packages/core/contracts/financial-templates/common/FeePayer.sol

```diff
@@ -100,31 +100,74 @@ abstract contract FeePayer is AdministrateeInterface, Testable, Lockable {
+    function payRegularFees() public nonReentrant() returns (FixedPoint.Unsigned memory) {
         uint256 time = getCurrentTime();
         FixedPoint.Unsigned memory collateralPool = _pfc();
 
+        // Fetch the regualar fees, late penatly and the max posible to pay given the current collateral within the contract.
+        (
+            FixedPoint.Unsigned memory regularFee,
+            FixedPoint.Unsigned memory latePenalty,
+            FixedPoint.Unsigned memory totalPaid
+        ) = getOutstandingRegularFees(time);
+        lastPaymentTime = time;
+
+        // If there are no fees to pay then exit early.
+        if (totalPaid.isEqual(0)) {
+            return totalPaid;
+        }
+
+        emit RegularFeesPaid(regularFee.rawValue, latePenalty.rawValue);
+
+        _adjustCumulativeFeeMultiplier(totalPaid, collateralPool);
+
+        if (regularFee.isGreaterThan(0)) {
+            StoreInterface store = _getStore();
+            collateralCurrency.safeIncreaseAllowance(address(store), regularFee.rawValue);
+            store.payOracleFeesErc20(address(collateralCurrency), regularFee);
+        }
+
+        if (latePenalty.isGreaterThan(0)) {
+            collateralCurrency.safeTransfer(msg.sender, latePenalty.rawValue);
+        }
+        return totalPaid;
+    }
+
+    /**
+     * @notice Fetch any regular fees that the contract has pending but has not yet paid. If the fees to be paid are more
+     * than the total collateral within the contract then the totalPaid returned is full contract collateral amount.
+     * @dev This returns 0 and exit early if there is no pfc, fees were already paid during the current block, or the fee rate is 0.
+     * @return regularFee outstanding unpaid regualr fee.
+     * @return latePenalty outstanding unpaid late fee for being late in previous fee payments.
+     * @return totalPaid Amount of collateral that the contract paid (sum of the amount paid to the Store and caller).
+     */
+    function getOutstandingRegularFees(uint256 time)
+        public
+        view
+        returns (
+            FixedPoint.Unsigned memory regularFee,
+            FixedPoint.Unsigned memory latePenalty,
+            FixedPoint.Unsigned memory totalPaid
+        )
+    {
+        StoreInterface store = _getStore();
+        FixedPoint.Unsigned memory collateralPool = _pfc();
+
         // Exit early if there is no collateral from which to pay fees.
         if (collateralPool.isEqual(0)) {
-            // Note: set the lastPaymentTime in this case so the contract is credited for paying during periods when it
-            // has no locked collateral.
-            lastPaymentTime = time;
-            return totalPaid;
+            return (regularFee, latePenalty, totalPaid);
         }
 
         // Exit early if fees were already paid during this block.
         if (lastPaymentTime == time) {
-            return totalPaid;
+            return (regularFee, latePenalty, totalPaid);
```

Comment:
Could this and the above if statement be combined to simplify:
```
// Exit early if there is no collateral or if fees were already paid during this block.
if (collateralPool.isEqual(0) || lastPaymentTime == time) {
   return (regularFee, latePenalty, totalPaid);
}
```

Reply:
yup can do & done!